chore: pin binary version in dockerfile

GabriFedi97 opened the pull request:
To avoid unexpected failures, due to possible breaking changes to the underlying dockerfile-template tool, it is better to pin the version to the one currently used.

We cannot start directly from the `ghcr.io/bossm8/dockerfile-templater:0.0.3` because it is a `FROM scratch` image and we need at least the shall sh and few other commands to execute the `entrypoint`.

GabriFedi97 committed:
chore: pin binary version in dockerfile
To avoid unexpected failures, due to possible
breaking changes to the underlying dockerfile-template tool,
it is better to pin the version to the one currently used.

Signed-off-by: Gabriele Fedi <gabriele.fedi@enterprisedb.com>

hh24k commented on Dockerfile:
```diff
@@ -1,4 +1,9 @@
-FROM ghcr.io/bossm8/dockerfile-templater:debug
```
can we just pin the tag as `0.03-debug`, whose difference from `0.03` is that it uses `busybox:musl` as base image instead of `scratch`?

hh24k replied:
https://github.com/bossm8/dockerfile-templater/pkgs/container/dockerfile-templater/138768373?tag=0.0.3-debug

fcanovai committed:
feat: make Dockerfile more solid

Avoid renaming the binary, stop failing scripts,
pin alpine version and avoid COPY to a relative
dir without WORKDIR defined.
